.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+on: push
+
+jobs:
+  code-quality:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: erlef/setup-beam@v1
+        with:
+          # Just some default versions to run code quality tools with.
+          # These can be upgraded as and when but probably won't need to change much.
+          otp-version: "24.0.6"
+          elixir-version: "1.12.3"
+      - uses: actions/cache@v2
+        with:
+          path: |
+            deps
+            _build
+            priv/plts
+          key: ${{ runner.os }}-mix-${{ hashFiles('**/mix.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-mix-
+      - run: mix deps.get
+      - run: mix credo --strict
+        continue-on-error: true
+      - run: mix format --check-formatted
+        continue-on-error: true
+      - run: mix dialyzer
+  test:
+    strategy:
+      matrix:
+        os: 
+          - ubuntu-latest
+        otp-version:
+          - "24.0.6"
+        elixir-version: 
+          - "1.12.3"
```

Comment:
We should probably setup an actual matrix  and test previous versions of Elixir and OTP as well.

Reply:
Yep.  Any suggestions for that?  I was starting simple so at least there is something in place as we go forward.